boto3/dynamodb/transform.py

```diff
@@ -0,0 +1,188 @@
+# Copyright 2015 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+# http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+from collections import Mapping
+import functools
+
+from boto3.dynamodb.types import TypeSerializer, TypeDeserializer
+from boto3.dynamodb.conditions import ConditionBase
+from boto3.dynamodb.conditions import ConditionExpressionBuilder
+
+
+def register_high_level_interface(base_classes, **kwargs):
+    # This assumes the base class is the base resource object
+    resource_base_class = base_classes[0]
+
+    class DynamoDBHighLevelResource(resource_base_class):
```

Comment:
The reason I did this was that I needed a handler to the event emitter. By creating the class in the handler, I get a handle to the base resource class and if I subclass from that and call ``super()`` on its ``__init__`` I will get an event emitter.

Ideally, I would add the event emitter to the event, but unfortunately, I cannot because the event emitter at that event is for the factory so if I register it to that emitter the higher level abstraction will be applied to all future clients that are created even though it is only called when the dynamodb resource is created.

I could also take it out of the method and subclass it from ``ServiceResource``, but it may need to be updated in the future if we ever build more base abstractions on top of ``ServiceResource``. However for now, this is completely viable way of doing it.